File: web/src/api/image.ts
import request from "../utils/request";

export const UploadImage = (image: File) => request({
  url: '/image/upload',
  method: 'post',
  data: {
    image,
  }
})

export const GetAllImage = () => request({
  url: '/image/list',
  method: 'get',
})

export const GetImageByAuthor = () => request({
  url: '/image/get',
  method: 'get',
})

export const DeleteImage = (id: number) => request({
  url: '/image/delete',
  method: 'delete',
  params: {
    id
  }
})

export const ImageCount = () => request({
  url: '/image/count',
  method: 'get'
})

File: server/router/image.go
package router

import (
	"github.com/gin-gonic/gin"
	"server/api"
)

func ImageRouter(router *gin.Engine) {
	imageApi := api.ImageApi{}

	image := router.Group("api/image")
	{
		image.POST("/upload", imageApi.UploadImage)
		image.GET("/list", imageApi.GetAllImage)
		image.GET("/get", imageApi.GetImageByAuthor)
		image.DELETE("/delete", imageApi.DeleteImage)
		image.GET("/count", imageApi.ImageCount)
	}

}
